front/app/modules/commercial/flag_inappropriate_content/admin/components/InappropriateContentWarning.tsx

```diff
@@ -23,17 +30,49 @@ const WarningIcon = styled(Icon)`
   width: 16px;
 `;
 
-interface Props {}
+interface Props {
+  inappropriateContentFlagId: string;
+}
 
 const InappropriateContentWarning = ({
+  inappropriateContentFlagId,
   intl: { formatMessage },
 }: Props & InjectedIntlProps) => {
-  return (
-    <Container>
-      <WarningIcon name="exclamation-trapezium" />
-      <WarningContent>{formatMessage(messages.warningText)} </WarningContent>
-    </Container>
+  const inappropriateContentFlag = useInappropriateContentFlag(
+    inappropriateContentFlagId
   );
+
+  if (!isNilOrError(inappropriateContentFlag)) {
+    const flagType = getFlagType(inappropriateContentFlag);
+    const reasonCode = inappropriateContentFlag.attributes.reason_code;
+
+    // if reasonCode is null, it means the flag has been removed
+    // and we shouldn't display anything
+    if (reasonCode && flagType) {
+      return (
+        <Container>
+          <WarningIcon name="exclamation-trapezium" />
+          <WarningContent>
+            {
+              {
+                nlp_flagged: formatMessage(messages.nlpFlaggedWarningText),
+                user_flagged: formatMessage(messages.userFlaggedWarningText, {
+                  reason: {
+                    inappropriate: formatMessage(messages.inappropriate),
+                    wrong_content: formatMessage(messages.wrong),
+                  }[reasonCode],
+                }),
+              }[flagType]
+            }
+          </WarningContent>
+        </Container>
+      );
+    }
+
+    return null;
```

Comment:
This is very minor but it might be a good idea to reverse this a bit -> get all the conditions that would return `null` out of the way first and then add the component. It's how we do it elsewhere and it's a bit more readable, imo.

Reply:
In fact, returning null at the end (as the fallback) was the convention, but then some of us started doing it the opposite way. ;-) in most of the codebase, we still do it like this.

Reply:
I personally find this clearer, because you explicitely say what you need in order for the component to render properly. But I guess it's what you're used to.